### PR TITLE
Frohro

### PR DESCRIPTION
**Pull request by frohro**
This is my work on #739, though I decided that the coordinate system wasn't really that important to me, so I just made snap to grid work on the rectangle, and arrow, as these are more important to me.  Making a coordinate system is easy with the arrow, and I can make ones with arrows on both ends, etc.  Snapping makes them look good.

### CODE REVIEW

**Comment by morrolinux** [comment]
Ok so let me summarize it:
1) The arrow snapping is only working on the arrow tip for me (not both sides) 
2) The snapping tolerance should be configurable. Until we do this, I would not change it for consistency since it's so drastic.
3) Tab needs to be fixed 

Overall seems fine! 
Thanks for the contribution :) 
I can merge it when the above is fixed

**Comment by frohro** [comment]
I'm not quite certain how to make it so you can have the first point off grid, but the arrow with snap to grid, so right now you can allow the arrow to not snap (by holding shift) and have the initial point snap to grid, but not the other way around.  Both ends snap to grid in that case.  

I don't have time to figure this out now.  Maybe later.  :-)

**Comment by morrolinux** [comment]
Let me show you what happens with the arrow's grid snapping on my computer:
snapping

**Comment by frohro** [comment]
It looks like the latest update didn't get included, i.e., you still have tolerance = gridsize/2.0; in the snapToGrid function in BaseStrokeHandler.  Did you do a new pull?

File: src/control/tools/BaseStrokeHandler.cpp [hunk 1]
```diff
@@ -54,16 +109,16 @@ bool BaseStrokeHandler::onMotionNotifyEvent(const PositionInputData& pos)
...
 	this->redrawable->repaintRect(stroke->getX(), stroke->getY(),
-			stroke->getElementWidth(), stroke->getElementHeight());
+								  stroke->getElementWidth(), stroke->getElementHeight());
 
 	drawShape(currentPoint, pos.isShiftDown());
 	
 	rect.add(stroke->boundingRect());
 	double w = stroke->getWidth();
 
 	redrawable->repaintRect(rect.x - w, rect.y - w,
```
Comment (morrolinux) [inline src/control/tools/BaseStrokeHandler.cpp]: just an observation: this could be on a single line

**Comment by morrolinux** [comment]
Oh sorry you were right. I wasn't following the branch, so `git pull` wasn't giving anything new.
I can merge this now!

**Pull request merged**